### PR TITLE
Fix for LIS-DEPLOY_SCENARIO Testcases

### PR DESCRIPTION
**Pull request by sireeshapotnuru**
1. Fix in line to compare the return value from function
2. Fix where the checkpoint is not found, but VM is trying to restore the same checkpoint
3. Fix for command to get Linux Distro Version
4. Added param runMaxAllowedTime, since Run-LinuxCmd is resulting in timeout while downloading/extracting LIS RPM

### CODE REVIEW

**Comment by sireeshapotnuru** [comment]
[LISAv2 Test Results Summary]
Test Run On           : 03/11/2019 07:44:45
VHD Under Test        : \\redmond\wsscfs\OSTC\LIS\VHD\Cloudbase\CentOS\CentOS_6.8_32bit\centos_6.8_32bit.vhdx
Total Test Cases      : 8 (8 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:57

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 LIS-DEPLOY-SCENARIO-1                                                             PASS                  5.7
        Install-LIS-Scenario-1 : PASS
    2 LIS-DEPLOY-SCENARIO-2                                                             PASS                  8.7
        Install-LIS-Scenario-2 : PASS
    3 LIS-DEPLOY-SCENARIO-3                                                             PASS                 8.88
        Install-LIS-Scenario-3 : PASS
    4 LIS-DEPLOY-SCENARIO-4                                                             PASS                 4.19
        Install-LIS-Scenario-4 : PASS
    5 LIS-DEPLOY-SCENARIO-5                                                             PASS                 4.85
        Install-LIS-Scenario-5 : PASS
    6 LIS-DEPLOY-SCENARIO-6                                                             PASS                 4.86
        Install-LIS-Scenario-6 : PASS
    7 LIS-DEPLOY-SCENARIO-7                                                             PASS                13.46
        Install-LIS-Scenario-7 :  PASS
    8 LIS-DEPLOY-SCENARIO-8                                                             PASS                 4.73
        Install-LIS-Scenario-8 : PASS


Logs can be found at F:\LISAv2\MK72\TestResults\2019-11-03-00-44-40-2815


03/11/2019 08:41:50 : [INFO ] Current Git Branch:  fix_lis_deploy_scenarios
03/11/2019 08:41:50 : [INFO ] Current Git Commit ID: fe30051ab8fe6a94cec76b6dce9d95f210ff2d8c

**Comment by iamshital** [comment]
We're verifying this PR against various distros.
**Do not merge, until further further notification.**

File: Libraries/HyperV.psm1 [hunk 1]
```diff
@@ -664,17 +664,25 @@ function Apply-HyperVCheckpoint {
...
+        if($?) {
+            Stop-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost -TurnOff -Force
+            Restore-VMSnapshot -Name $CheckpointName -VMName $VM.RoleName -ComputerName $VM.HyperVHost -Confirm:$false
+            $msg = ("VM:{0} restored to checkpoint: {1}" `
                  -f ($VM.RoleName,$CheckpointName))
-        Write-LogInfo $msg
-        Start-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost
+            Write-LogInfo $msg
+            Start-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost
+            return $true
```
Comment (lizzha) [inline Libraries/HyperV.psm1]: this will return on the first VM, which changes the behavior of multiple VM scenarios

**Review by lizzha — CHANGES_REQUESTED** [review]

File: Libraries/HyperV.psm1 (hunk 1 above)
Comment (sireeshapotnuru) [reply Libraries/HyperV.psm1]: moved this line to end of the function which will return $true if multiple VMs provided are having same  checkpoint

**Review by lizzha — APPROVED** [review]

File: Libraries/HyperV.psm1 [hunk 2]
```diff
@@ -664,17 +664,25 @@ function Apply-HyperVCheckpoint {
         $VMData,
         [string]$CheckpointName
     )
-
     foreach ($VM in $VMData) {
-        Stop-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost -TurnOff -Force
-        Restore-VMSnapshot -Name $CheckpointName -VMName $VM.RoleName -ComputerName $VM.HyperVHost -Confirm:$false
-        $msg = ("VM:{0} restored to checkpoint: {1}" `
+        Get-VMSnapshot -Name $CheckpointName -VMName $VM.RoleName -ErrorAction Ignore | out-null
+        if($?) {
```
Comment (chvalean) [inline Libraries/HyperV.psm1]: style checks please:
if (...) {
} else {

File: Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1 [hunk 3]
```diff
@@ -365,10 +374,10 @@ Function Install-LIS-Scenario-6 ($PreviousTestResult, $LISTarballUrlOld, $LISTar
 # Expected result : Verify that LIS has upgraded to Current LIS successfully after kernel upgrade
 Function Install-LIS-Scenario-7 ($PreviousTestResult, $LISTarballUrlOld, $LISTarballUrlCurrent) {
     if ($PreviousTestResult -eq "PASS") {
-        $is_oracle = Run-LinuxCmd -ip $allVMData.PublicIP -port $allVMData.SSHPort -username "root" -password $password -command "cat /etc/os-release | grep -i oracle" -ignoreLinuxExitCode:$true
-        if ($is_oracle) {
+        $detectedDistro = Detect-LinuxDistro -VIP $allVMData.PublicIP -SSHport $allVMData.SSHPort -testVMUser $user -testVMPassword $password
```
Comment (chvalean) [inline Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1]: you don't need this, detectDistro is a global var and you can use it directly from there.

**Review by chvalean — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1 (hunk 3 above)
Comment (sireeshapotnuru) [reply Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1]: Removed calling again Detect-LinuxDistro function and directly used detectDistro

File: Libraries/HyperV.psm1 (hunk 2 above)
Comment (sireeshapotnuru) [reply Libraries/HyperV.psm1]: added space after if

File: Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1 [hunk 4]
```diff
@@ -33,7 +33,7 @@ Function Upgrade-LIS ($LISTarballUrlOld, $LISTarballUrlCurrent, $allVMData , $Te
         $CurrentLISExtractCommand = "rm -rf LISISO^wget $($LISTarballUrlCurrent)^tar -xzf $($LISTarballUrlCurrent | Split-Path -Leaf)^cp -ar LISISO/* ."
         $LISExtractCommands = $CurrentLISExtractCommand.Split("^")
         foreach ( $LISExtractCommand in $LISExtractCommands ) {
-            Run-LinuxCmd -username "root" -password $password -ip $allVMData.PublicIP -port $allVMData.SSHPort -command $LISExtractCommand
+            Run-LinuxCmd -username "root" -password $password -ip $allVMData.PublicIP -port $allVMData.SSHPort -command $LISExtractCommand -runMaxAllowedTime 20000
```
Comment (chvalean) [inline Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1]: the value is too high, that's over 5 hours for the extract command? please fix all 20.000 occurrences

**Review by chvalean — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1 (hunk 4 above)
Comment (sireeshapotnuru) [reply Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1]: changed to -runMaxAllowedTime 2000 -maxRetryCount 3

**Review by chvalean — APPROVED** [review]

**Review by chvalean — APPROVED** [review]

**Comment by chvalean** [comment]
@iamshital please approve this if all testing has been completed with these changes, so we can merge it.

File: Libraries/HyperV.psm1 [hunk 5]
```diff
@@ -666,17 +666,25 @@ function Apply-HyperVCheckpoint {
         $VMData,
         [string]$CheckpointName
     )
-
     foreach ($VM in $VMData) {
-        Stop-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost -TurnOff -Force
-        Restore-VMSnapshot -Name $CheckpointName -VMName $VM.RoleName -ComputerName $VM.HyperVHost -Confirm:$false
-        $msg = ("VM:{0} restored to checkpoint: {1}" `
+        Get-VMSnapshot -Name $CheckpointName -VMName $VM.RoleName -ErrorAction Ignore | out-null
```
Comment (LiliDeng) [inline Libraries/HyperV.psm1]: + -ComputerName?

Comment (sireeshapotnuru) [reply Libraries/HyperV.psm1]: Added param ComputerName

**Pull request merged**